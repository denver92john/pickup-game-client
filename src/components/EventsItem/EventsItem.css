.event-item {
    margin-bottom: 5px;
    background-color: white;
    padding: 10px;
    text-align: left;
}

.event-item-header {
    display: flex;
    justify-content: space-between;
    padding-bottom: 10px;
}

.event-item-footer {
    padding-bottom: 5px;
}

/*.event-item-title {
    display: inline-block;
}

.event-item-sport {
    display: inline-block;
}*/

.event-item-datetime {

}