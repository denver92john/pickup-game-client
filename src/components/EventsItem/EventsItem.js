import React, {Component} from 'react';
import moment from 'moment';
import {Link} from 'react-router-dom';
import './EventsItem.css';

export default class EventsItem extends Component {
    render() {
        const {event} = this.props;
        let datetime = moment(event.datetime).format('ddd, MMM-DD-YYYY h:mm a');
        return (
            <li className="event-item">
                <Link to={`/event/${event.id}`} className="events-item">
                    <header className="event-item-header">
                        <h4 className="event-item-title">{event.title}</h4>
                        <p className="event-item-sport">{event.sport}</p>
                    </header>
                    <footer className="event-item-footer">
                        <p className="event-item-datetime">When: {datetime}</p>
                    </footer>
                </Link>
            </li>
            
        );
    }
}