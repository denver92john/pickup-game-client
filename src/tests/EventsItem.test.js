import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import EventsItem from '../components/EventsItem/EventsItem';

describe(`EventsItem component`, () => {
    const eventProp = {
        datetime: '2019-11-15T16:20:55.000Z',
        description: 'Thanksgiving football game',
        host: {
            first_name: 'John',
            id: 1,
            last_name: 'Denver',
            username: 'JDenver'
        },
        id: 1,
        max_players: 10,
        number_of_players: 2,
        player_id: '',
        sport: 'football',
        title: 'Turkey Bowl'
    }

    it(`renders without errors`, () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <EventsItem event={eventProp} />
            </BrowserRouter>,
            div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it(`renders UI as expected`, () => {
        const tree = renderer
            .create(
                <BrowserRouter>
                    <EventsItem event={eventProp} />
                </BrowserRouter>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})