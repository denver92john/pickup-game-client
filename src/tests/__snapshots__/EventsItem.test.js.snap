// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventsItem component renders UI as expected 1`] = `
<li
  className="event-item"
>
  <a
    className="events-item"
    href="/event/1"
    onClick={[Function]}
  >
    <header
      className="event-item-header"
    >
      <h4
        className="event-item-title"
      >
        Turkey Bowl
      </h4>
      <p
        className="event-item-sport"
      >
        football
      </p>
    </header>
    <footer
      className="event-item-footer"
    >
      <p
        className="event-item-datetime"
      >
        <span
          className="black"
        >
          When:
        </span>
         
        Fri, Nov 15, 2019 @ 11:20 AM
      </p>
    </footer>
  </a>
</li>
`;
